polar-core/src/vm.rs

```diff
@@ -530,6 +536,16 @@ impl PolarVirtualMachine {
             }
             .into());
         }
+        // For LookupExternal and NextExternal goals, make sure that the call id result variable is unbound
+        match goal {
+            Goal::LookupExternal { call_id, .. } | Goal::NextExternal { call_id, .. } => {
+                assert!(matches!(
+                    self.variable_state(self.get_call_sym(&call_id)),
+                    VariableState::Unbound
+                ));
```

Comment:
would be good to add an explanation as the second arg to this `assert!()` macro; once you've done that, the comment on 539 might be redundant